File: backend/src/main/java/com/handmade/store/service/ReviewService.java
package com.handmade.store.service;

import com.handmade.store.dto.common.PageResponse;
import com.handmade.store.dto.review.ReviewRequest;
import com.handmade.store.dto.review.ReviewResponse;

import java.util.Map;

public interface ReviewService {

    ReviewResponse addReview(String email, Long productId, ReviewRequest request);

    PageResponse<ReviewResponse> getReviewsByProduct(Long productId, int page, int size);

    ReviewResponse updateReview(Long id, String email, ReviewRequest request);

    Map<String, Object> deleteReview(Long id, String email);

    Map<String, Object> deleteReviewAsAdmin(Long id);

    Map<String, Object> canReview(String email, Long productId);

    PageResponse<ReviewResponse> getAllReviews(int page, int size);
}

File: frontend/src/services/reviewService.js
import api from './api';

export const getReviewsByProduct = (productId, params) => {
  return api.get(`/reviews/product/${productId}`, { params });
};

export const addReview = (productId, data) => {
  return api.post(`/reviews/product/${productId}`, data);
};

export const updateReview = (id, data) => {
  return api.put(`/reviews/${id}`, data);
};

export const deleteReview = (id) => {
  return api.delete(`/reviews/${id}`);
};

export const canReview = (productId) => {
  return api.get(`/reviews/product/${productId}/can-review`);
};
